detox/src/utils/XCUITestRunner.js

```diff
@@ -0,0 +1,51 @@
+const DetoxRuntimeError = require('../errors/DetoxRuntimeError');
+
+const { execWithRetriesAndLogs } = require('./childProcess/exec');
+const environment = require('./environment');
+
+class XCUITestRunner {
+    constructor({ simulatorId }) {
+        this.simulatorId = simulatorId;
+    }
+
+    async execute(invocationParams) {
+        const base64InvocationParams = Buffer.from(JSON.stringify(invocationParams)).toString('base64');
+
+        const runnerPath = await environment.getXCUITestRunnerPath();
+        if (!runnerPath) {
+            throw new DetoxRuntimeError({
+                message: 'XCUITest runner path could not be found',
+                hint: DetoxRuntimeError.reportIssue,
+            });
+        }
+
+        const flags = [
+            '-xctestrun', runnerPath,
+            '-sdk', 'iphonesimulator',
+            '-destination', `"platform=iOS Simulator,id=${this.simulatorId}"`,
+            'test-without-building',
+        ];
+
+        try {
+            return await execWithRetriesAndLogs(
```

Comment:
`TEST_RUNNER_PARAMS` isn't recognized when passing from `env` in options, it might be because of a limitation or specific handling of environment variables by `xcodebuild`.

Reply:
btw, they also have specific requirement that all relevant environment variables must start with `TEST_RUNNER_`, and any variables not adhering to this naming convention may not be passed through 🤷🏼‍♂️

Reply:
> If you originally meant to keep some payload details in logs for later troubleshooting, I'd advise just to log your payload separately as a plain text via our logger – this way or that way, reading plain text is nicer than decoding Base64 with eyes in the logs. :)

Done